src/analyticsService.ts

```diff
@@ -1,93 +1,147 @@
-import NavigationService from './navigationService';
+// Modules
+import { ApplicationInsights } from '@microsoft/applicationinsights-web';
+import { ReactNativePlugin } from '@microsoft/applicationinsights-react-native';
 
-// let instrumentationKey:string = '';
-let appInsights: any = null;
-let allowTracking: boolean = false;
-let commonTrackingProperties: object = {};
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type {
+  IAppInsights,
+  IPageViewTelemetry,
+  IAutoExceptionTelemetry,
+  ICustomProperties,
+  IEventTelemetry,
+  IExceptionTelemetry,
+  IConfig,
+} from '@microsoft/applicationinsights-web';
 
-// export const setInstrumentationKey = (keyValue:string): void => {
-//     instrumentationKey = keyValue;
-// }
+// Interfaces
+interface IParams {
+  apiKey: string;
+  commonProps?: { [key: string | number]: any };
+  config?: IConfig;
+}
+export interface IAnalyticsInstance {
+  enable: () => void;
+  disable: () => void;
+  trackPageView: (pageName: string, props: any) => void;
+  trackException: (exception: any, props: any) => void;
+  trackEvent: (event: any, props: any) => void;
+  instance: IAppInsights;
```

Comment:
Why the separation between the `trackPageView` and `trackEvent`?  What are the differences between both?

Reply:
The trackPageView method can accept additional page view event data whereas the trackEvent can only accept custom properties.

Both these methods are present in and exposed from app insights itself